jsonexample/test/simple_object_unit_test.dart

```diff
@@ -0,0 +1,196 @@
+    'aListOfStrings': [],
+    'aListOfInts': [],
+    'aListOfDoubles': []
+  };
+
+  const Map<String, dynamic> unexpectedPropertyjson = {
+    'aString': 'Blah, blah, blah.',
+    'anInt': 1,
+    'aDouble': 1.0,
+    'aListOfStrings': ['one', 'two', 'three'],
+    'aListOfInts': [1, 2, 3],
+    'aListOfDoubles': [1.0, 2.0, 3.0],
+    'unexpectedProperty': 'Whoops!'
+  };
+
+  group('ConvertedSimpleObject unit tests', () {
+    test('Typical object', () {
+      final simpleObject = ConvertedSimpleObject.fromJson(typicalObjectJson);
+
+      expect(simpleObject, isNotNull);
+      expect(simpleObject.aString, "Blah, blah, blah.");
+      expect(simpleObject.anInt, 1);
+      expect(simpleObject.aDouble, 1.0);
+      expect(simpleObject.aListOfStrings, ['one', 'two', 'three']);
+      expect(simpleObject.aListOfInts, [1, 2, 3]);
+      expect(simpleObject.aListOfDoubles, [1.0, 2.0, 3.0]);
+    });
+
+    test('Empty object', () {
+      final simpleObject = ConvertedSimpleObject.fromJson(<String, dynamic>{});
+
+      expect(simpleObject, isNotNull);
+      expect(simpleObject.aString, isNull);
+      expect(simpleObject.anInt, isNull);
+      expect(simpleObject.aDouble, isNull);
+      expect(simpleObject.aListOfStrings, isNull);
+      expect(simpleObject.aListOfInts, isNull);
+      expect(simpleObject.aListOfDoubles, isNull);
+    });
+
+    test('Empty lists', () {
+      final simpleObject = ConvertedSimpleObject.fromJson(emptyListJson);
+
+      expect(simpleObject, isNotNull);
+      expect(simpleObject.aString, "Blah, blah, blah.");
+      expect(simpleObject.anInt, 1);
+      expect(simpleObject.aDouble, 1.0);
+      expect(simpleObject.aListOfStrings, []);
+      expect(simpleObject.aListOfInts, []);
+      expect(simpleObject.aListOfDoubles, []);
+    });
+
+    test('Extra properties', () {
+      final simpleObject =
+          ConvertedSimpleObject.fromJson(unexpectedPropertyjson);
+
+      expect(simpleObject, isNotNull);
+      expect(simpleObject.aString, "Blah, blah, blah.");
+      expect(simpleObject.anInt, 1);
+      expect(simpleObject.aDouble, 1.0);
+      expect(simpleObject.aListOfStrings, ['one', 'two', 'three']);
+      expect(simpleObject.aListOfInts, [1, 2, 3]);
+      expect(simpleObject.aListOfDoubles, [1.0, 2.0, 3.0]);
+    });
+  });
+
+  group('SerializableSimpleObject unit tests', () {
+    test('Typical object', () {
+      final simpleObject = SerializableSimpleObject.fromJson(typicalObjectJson);
+
+      expect(simpleObject, isNotNull);
+      expect(simpleObject.aString, "Blah, blah, blah.");
+      expect(simpleObject.anInt, 1);
+      expect(simpleObject.aDouble, 1.0);
+      expect(simpleObject.aListOfStrings, ['one', 'two', 'three']);
+      expect(simpleObject.aListOfInts, [1, 2, 3]);
+      expect(simpleObject.aListOfDoubles, [1.0, 2.0, 3.0]);
+    });
+
+    test('Empty object', () {
+      final simpleObject =
+          SerializableSimpleObject.fromJson(<String, dynamic>{});
```

Comment:
[nit] Why not have this empty map among the constants as well?

Reply:
Originally it was just the brackets {}, so it felt silly. Makes more sense now, though, so fixed.